Entity/CategoryManager.php

```diff
@@ -143,11 +143,11 @@ public function getRootCategory($context = null)
     /**
      * NEXT_MAJOR: add this method to the interface.
      *
-     * @param ContextInterface $context
+     * @param ContextInterface|string|null $context
      *
      * @return CategoryInterface[]
      */
-    public function getRootCategoriesForContext(ContextInterface $context = null)
+    public function getRootCategoriesForContext($context = null)
```

Comment:
You wanted to make these methods final, but we didn't do it, because they were stubbed in tests.
I really don't think, that anyone has overridden this method after the latest release. But if it's critical, I can change it to master branch.